File: test/org/mozilla/services/json/TestExJSONTuple.java
package org.mozilla.services.json;
import org.junit.Test;
import static org.junit.Assert.assertEquals;


import org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorFactory;
import org.mozilla.services.json.ExJSONTuple;
import org.apache.hadoop.hive.ql.udf.generic.UDTFCollector;
import org.apache.hadoop.hive.ql.exec.UDTFOperator;
import java.util.ArrayList;
import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector;

public class TestExJSONTuple
{
    public TestExJSONTuple() {};

    public ArrayList run_jsontuple(String[] user_args) throws Exception {
        int i = 0;

        ArrayList<String> result = new ArrayList<String>();

        ExJSONTuple func = new ExJSONTuple();
        func.setCollector(new UDTFCollector(new UDTFOperator()));

        ArrayList<ObjectInspector> args = new ArrayList<ObjectInspector>();

        for (i = 0; i < user_args.length; i++) {
            args.add(PrimitiveObjectInspectorFactory.writableStringObjectInspector);
        }
        ObjectInspector[] args_arr = new ObjectInspector[args.size()];
        args_arr = args.toArray(args_arr);
        func.initialize(args_arr);

        ArrayList<Object> func_args = new ArrayList<Object>();
        for (i = 0; i < user_args.length; i++)
        {
            func_args.add(PrimitiveObjectInspectorFactory.writableStringObjectInspector.create(user_args[i]));
        }

        func.process(func_args.toArray());

        for (i = 0; i < func.retCols.length;i++) {
            result.add(func.retCols[i].toString());
        }
        return result;
    }

    @Test
    public void test_exjsontuple() throws Exception
    {

        String args[] = new String[]{"{\"foo\": 42, \"bar\": 55}", "foo", "bar"};

        ArrayList result = run_jsontuple(args);
        assertEquals(2, result.size());
        assertEquals("42", result.get(0).toString());
        assertEquals("55", result.get(1).toString());
    }

}

File: test_exjsontuple.py
import os
import sys
import StringIO

for file in os.listdir('.'):
    sys.path.append(file)
sys.path.append('../dist/lib/MetlogHive.jar')


from java.util import ArrayList
from org.apache.hadoop.io import Text
from org.apache.hadoop.hive.serde2.objectinspector.primitive import *
from org.apache.hadoop.hive.serde2.objectinspector import *
from org.mozilla.services.json import *
from org.apache.hadoop.hive.ql.udf.generic import UDTFCollector
from org.apache.hadoop.hive.ql.exec import UDTFOperator

func = ExJSONTuple()
op = UDTFOperator()
collector = UDTFCollector(op)
func.setCollector(collector)

values = []
values.append(PrimitiveObjectInspectorFactory.writableStringObjectInspector)
values.append(PrimitiveObjectInspectorFactory.writableStringObjectInspector)
values.append(PrimitiveObjectInspectorFactory.writableStringObjectInspector)

#op.initialize(func.conf, values)

func.initialize(values)
args = []

json_txt = PrimitiveObjectInspectorFactory.writableStringObjectInspector.create('{"foo": 42, "bar": 55}')
foo_txt = PrimitiveObjectInspectorFactory.writableStringObjectInspector.create('foo')
bar_txt = PrimitiveObjectInspectorFactory.writableStringObjectInspector.create('bar')

stderr = sys.stderr
stdout = sys.stdout

mock_stderr = StringIO.StringIO()
mock_stdout = StringIO.StringIO()

sys.stdout = mock_stdout
sys.stderr = mock_stderr

try:
    func.process([json_txt, foo_txt, bar_txt])
except:
    print '========== Std Err ========='
    print mock_stderr.getvalue()
    print '========== Std Out ========='
    print mock_stdout.getvalue()
    print '-----------'
finally:
    sys.stderr = stderr
    sys.stdout = stdout

print 'Done!!!!!'

